Source/Core/Core/NetPlayServer.cpp

```diff
@@ -1551,72 +1584,117 @@ void NetPlayServer::AbortGameStart()
 }
 
 // called from ---GUI--- thread
-bool NetPlayServer::SyncSaveData()
+bool NetPlayServer::CollectSaveSyncInfo(SaveSyncInfo* sync_info)
 {
-  // We're about to sync saves, so set m_saves_synced to false (waits to start game)
-  m_saves_synced = false;
-
-  m_save_data_synced_players = 0;
-
   u8 save_count = 0;
-
   for (ExpansionInterface::Slot slot : ExpansionInterface::MEMCARD_SLOTS)
   {
     if (m_settings.exi_device[slot] == ExpansionInterface::EXIDeviceType::MemoryCard ||
         Config::Get(Config::GetInfoForEXIDevice(slot)) ==
             ExpansionInterface::EXIDeviceType::MemoryCardFolder)
     {
-      save_count++;
+      ++save_count;
     }
   }
 
-  const auto game = m_dialog->FindGameFile(m_selected_game_identifier);
+  auto game = m_dialog->FindGameFile(m_selected_game_identifier);
   if (game == nullptr)
   {
     PanicAlertFmtT("Selected game doesn't exist in game list!");
     return false;
   }
 
-  bool wii_save = false;
+  bool has_wii_save = false;
```

Comment:
Seems like this is basically a duplicate of SaveSyncInfo's fields. Maybe consider changing this function to return an `std::optional<SaveSyncInfo>`? Then you could have a single SaveSyncInfo variable and return that and not have to move things manually.

Reply:
Yeah I had that originally but the compiler yelled at me for forward-declaring the type inside the optional, even if it's just used as a return value of a member function. I'll see what I can do though.

Reply:
Hmm I think that should definitely work because the return type of a function *declaration* does not have to be complete, and that means the class template shouldn't be getting implicitly instantiated at the function declaration point ([[temp.inst]/2](https://timsong-cpp.github.io/cppwp/n4868/temp.inst#2)).

Is this similar to what you had? https://godbolt.org/z/hP847rh9o

Reply:
Okay yeah trying it again it does work, not sure what I did wrong before. That's definitely better.